packages/client/src/classes/orchestrator.ts

```diff
@@ -0,0 +1,84 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+
+import { NodeResponseStatus, StatusType } from '~enums'
+import { NodeRequest, StatusResponse } from '~interfaces'
+import {
+	executeNode,
+	genericCheckStatus,
+	returnOrchestratorStatus,
+	terminateRun,
+} from '~resources'
+import { isStatusProcessing, wait } from '~utils'
+
+export class Orchestrator {
+	public _onCancel: ((...args) => void) | undefined
+	public _onStart: ((...args) => void) | undefined
+	public _onUpdate: ((...args) => void) | undefined
+	public _onComplete: ((...args) => void) | undefined
+	private _orchestratorResponse
+	static instance: Orchestrator
+
+	constructor() {
```

Comment:
So this becomes a true singleton here - is there any potential conflict with the two implementations that use it overwriting the handler methods? You may want to consider a factory function instead that can manage instances (the factory function itself would be a singleton, but could provide new copies of the orchestrators as needed). You can also implement a singleton in JavaScript by using module-level variables. Thematic has an example of a [singleton context](https://github.com/microsoft/thematic/blob/main/packages/react/src/provider/ThematicContext.ts).